.github/workflows/main.yml

```diff
@@ -74,6 +74,35 @@ jobs:
           retention-days: 1
           if-no-files-found: error
 
+  run-cwl-auto-conda:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: checkout-repository
+        uses: actions/checkout@v2
+
+      - name: install-basic-deps
+        uses: ./.github/actions/install-basic-deps
+
+      - name: run-workflow
+        shell: bash -l {0}
+        run: |
+          sudo apt-get update && sudo apt-get install --no-install-recommends --yes python3-pip
+          wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
+          bash Miniconda3-latest-Linux-x86_64.sh -b -p conda
+          conda/bin/conda init
+          source ~/.bashrc
+          sudo python3 -m pip install cytoolz conda cwltool[deps]
```

Comment:
@mr-c, so far I wasn't able to fix this.. 

The first problem was that cwltool seems to want to use the conda python package at some point for package management, but it wasn't available. So I added the manual installation of conda from the installer, as well as the pip install of conda (seemed to also be required, shouldn't that be installed with `cwltool[deps]`). Then, an error from conda appeared that cytoolz was not available (maybe it is missing as requirement to their setup?). But now there is yet another error... I would guess it is some versioning issue, maybe we need to update conda to a newer version? At least in the log conda says that there is a newer version available...

Reply:
I tried to make a Dockerfile to reproduce this locally, but I failed. There is some GitHub Actions worker configuration that is causing a problem:

``` Dockerfile
FROM docker.io/ubuntu:focal-20220426

RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
	gcc \
	python3-dev \
	python3-pip \
	nodejs \
	curl \
	libgl1

RUN pip3 install cwltool[deps]

WORKDIR /tmp
COPY . .

WORKDIR /tmp/simple_use_case/cwl
RUN python3 -m cwltool --debug --beta-conda-dependencies wf_run_use_case.cwl
```

Reply:
(that is to say, the above `Dockerfile` causing the workflow to run successfully.)

Reply:
hm, for me it actually fails, although with a different error: gmsh is not found... I'll see what I can do. Thanks for the Dockerfile in any case!!

Reply:
> hm, for me it actually fails, although with a different error: gmsh is not found... I'll see what I can do. Thanks for the Dockerfile in any case!!

I'm surprised to hear that! Here is a log of running `docker build .` in a checkout of this branch: https://gist.github.com/mr-c/0f6d514358d0448c402c813fa90b705e

Reply:
nevermind, I ran it in the wrong folder last night.. That folder was not on your branch. It works now, thanks! Sorry about that!

Reply:
@mr-c, I tried a bunch of things which unfortunately all didn't work. Github Actions and Conda seems difficult :). I finally added the setup-miniconda action (but here without environment specification) to the actions so that they take care of setting up conda appropriately. It now works, but I would like to clean up the Git history, removing all my suggestions clutter.

Reply:
Ideally, I think we would just squash everything after 9eb4f84b7648c4b24e023bf62fd91ea9ec0a7268 into it.

Reply:
@dglaeser Thanks for your perseverance! Feel free to force push and otherwise re-write the git history; maybe with a `Co-authored-by: Michael R. Crusoe <michael.crusoe@gmail.com>` if you go down to one commit

Reply:
Or I can do the squashing, if you'd prefer

Reply:
Regarding co-authorship. I squashed my stuff into your commit, that's fine for me.